### DISCUSSION TITLE
httpx async api misbehaving

### DISCUSSION THREAD

**Discussion by onlinejudge95** [encode/httpx#1638] [Q&A]
My use case requires a FastAPI server that acts as an API-Gateway. For an endpoint of `/api/email` the gateway sends a request to one of my services which send out an email. Given the nature of my service, I decide to use httpx for sending the request in an async manner. But I receive the following error
```
api-gateway_1  | Traceback (most recent call last):
api-gateway_1  |   File "/usr/src/app/src/routers/email/routes.py", line 25, in send_mail
api-gateway_1  |     response = await post(url=url, json=data, headers=headers)
api-gateway_1  | TypeError: object Response can't be used in 'await' expression
```
Here is my route definition
```
from httpx import post
...
@router.post("/")
async def send_mail(email: Email) -> Dict[str, str]:
    """
    API Gateway endpoint to send request to email service on /api/email route.
    """
    try:
        url = f"{settings.EMAIL_MICROSERVICE_URL}/api/email/"
        data = jsonable_encoder(email)
        headers = {"Accept": "application/json", "Content-Type": "application/json"}
        # response = await AsyncClient.post(url, json=data, headers=headers)
        response = await post(url=url, json=data, headers=headers)
        print(response.status_code)
    except Exception as e:
        print_exc()
        return {"status": "error", "message": str(e)}
    return {"status": "ok"}
```
Whereas removing the `await` keyword makes the request go through. Am I missing something?

**Comment by HarrySky** [comment]
From first page of docs "For more advanced topics, see the Advanced Usage section, the async support section"

Check this: https://www.python-httpx.org/async/

**Reply by onlinejudge95** [reply]
Uggghhhh, my bad this does not even need a discussion I already had the Asyncclient code ready with me just that I forgot to use it inside the `async with` anyways thanks for the time, i will be using coffee from now on for late-night coding.
